File: is52038b-labs/09/cpp/StringMatchCount.cpp
#include <iostream>

#include "StringMatch.hpp"
#include "NaiveStringMatch.hpp"
#include "RKStringMatch.hpp"
#include "KMPStringMatch.hpp"

int main() {
  std::string P = "aaaab";
  std::string T = "";
  StringMatch *matcher = new KMPStringMatch();

  matcher->match(T, P);
  std::cout << "char reads: " << matcher->counter.report() << std::endl;;

  delete matcher;
  return 0;
}

File: is52038b-labs/09/java/StringMatchCount.java
class StringMatchCount {
    public static void main(String args[]) {
	String P = "ab";
	String T = "aaaaaaaaaa";
	StringMatch matcher = new NaiveStringMatch();

	matcher.match(T, P);
	System.out.println("char reads: " + matcher.counter.report());
    }
}
